src/scripts/chat-message-response.js

```diff
@@ -22,11 +22,20 @@ async function triggerResponse(actor, request) {
     let alias = await actor.getFlag("unkenny", "alias");
     request = replaceAlias(request, alias, name);
 
+    let preamble = await actor.getFlag("unkenny", "preamble") || "";
     let parameters = await getGenerationParameters(actor);
     if (!parameters) {
         return;
     }
 
+    if (await actor.getFlag("unkenny", "includeBiography") && parameters.biography.trim()) {
+        preamble += `\n\n${parameters.biography}`;
+    }
+
+    console.log("Constructed Preamble:", preamble);
+
+    request = `${preamble}\n\n${request}`;
+
```

Comment:
Also, `messagesOrganisedForTemplate` currently throws an error if the preamble is falsy (e.g. if it is empty), but with this feature I don't think this should necessarily be an error any more. The logic is more complicated now.
(The function resides in collecting-chat-messages.js, but I think I can only comment files that changed with this PR.)

Reply:
This should be fixed in this commit but I'll have to split it from the other changes I made for the conversation mechanics.